### ISSUE TITLE
Intermittent UI test failure - org.mozilla.fenix.ui.TabbedBrowsingTest -  verifyUndoSnackBarTest

### ISSUE THREAD

**Issue by Archaeopteryx** [mozilla-mobile/fenix#22640]
### Firebase Test Run:
[Provide a Firebase test run report link here showcasing the problem](https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7673210351727308073/executions/bs.7aba5491cec86240/testcases/1/test-cases)
### Stacktrace:
```
androidx.test.uiautomator.UiObjectNotFoundException: UiSelector[TEXT=UNDO, RESOURCE_ID=org.mozilla.fenix.debug:id/snackbar_btn]
	at androidx.test.uiautomator.UiObject.click(UiObject.java:416)
	at org.mozilla.fenix.ui.robots.TabDrawerRobot.snackBarButtonClick(TabDrawerRobot.kt:157)
	at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$3.invoke(TabbedBrowsingTest.kt:188)
	at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$3.invoke(TabbedBrowsingTest.kt:184)
	at org.mozilla.fenix.ui.robots.BrowserRobot$Transition.openTabDrawer(BrowserRobot.kt:597)
	at org.mozilla.fenix.ui.TabbedBrowsingTest.verifyUndoSnackBarTest(TabbedBrowsingTest.kt:184)
	at java.lang.reflect.Method.invoke(Native Method)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
	at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
	at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
	at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:395)
	at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
```
### Build: 
https://github.com/mozilla-mobile/fenix/commit/4d9a58c156144f5cdda08972614d3febdfa98fdf

**Comment by AndiAJ** [comment]
Failed 1x on 12/1 Main 

[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7673210351727308073/executions/bs.7aba5491cec86240/testcases/1/test-cases)

```
androidx.test.uiautomator.UiObjectNotFoundException: UiSelector[TEXT=UNDO, RESOURCE_ID=org.mozilla.fenix.debug:id/snackbar_btn]
at androidx.test.uiautomator.UiObject.click(UiObject.java:416)
at org.mozilla.fenix.ui.robots.TabDrawerRobot.snackBarButtonClick(TabDrawerRobot.kt:157)
at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$3.invoke(TabbedBrowsingTest.kt:188)
at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$3.invoke(TabbedBrowsingTest.kt:184)
at org.mozilla.fenix.ui.robots.BrowserRobot$Transition.openTabDrawer(BrowserRobot.kt:597)
at org.mozilla.fenix.ui.TabbedBrowsingTest.verifyUndoSnackBarTest(TabbedBrowsingTest.kt:184)
at java.lang.reflect.Method.invoke(Native Method)
```

[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7673210351727308073/executions/bs.57dc46752a97e405/testcases/1/test-cases)

```
java.lang.AssertionError
at org.junit.Assert.fail(Assert.java:87)
at org.junit.Assert.assertTrue(Assert.java:42)
at org.junit.Assert.assertTrue(Assert.java:53)
at org.mozilla.fenix.ui.robots.BrowserRobot.verifyTabCounter(BrowserRobot.kt:111)
at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$4.invoke(TabbedBrowsingTest.kt:192)
at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$4.invoke(TabbedBrowsingTest.kt:191)
at org.mozilla.fenix.ui.robots.BrowserRobotKt.browserScreen(BrowserRobot.kt:658)
at org.mozilla.fenix.ui.TabbedBrowsingTest.verifyUndoSnackBarTest(TabbedBrowsingTest.kt:191)
at java.lang.reflect.Method.invoke(Native Method)
```

**Comment by AndiAJ** [comment]
Failed 1x on 12/2 Main 

[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/4646033380125252318/executions/bs.a2ff820cb8d3cc04/testcases/1/test-cases)

```
androidx.test.uiautomator.UiObjectNotFoundException: UiSelector[TEXT=UNDO, RESOURCE_ID=org.mozilla.fenix.debug:id/snackbar_btn]
at androidx.test.uiautomator.UiObject.click(UiObject.java:416)
at org.mozilla.fenix.ui.robots.TabDrawerRobot.snackBarButtonClick(TabDrawerRobot.kt:157)
at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$3.invoke(TabbedBrowsingTest.kt:188)
at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$3.invoke(TabbedBrowsingTest.kt:184)
at org.mozilla.fenix.ui.robots.BrowserRobot$Transition.openTabDrawer(BrowserRobot.kt:597)
at org.mozilla.fenix.ui.TabbedBrowsingTest.verifyUndoSnackBarTest(TabbedBrowsingTest.kt:184)
at java.lang.reflect.Method.invoke(Native Method)
```

[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/4646033380125252318/executions/bs.bc667e8b46a0c78e/testcases/1/test-cases)

```
java.lang.AssertionError
at org.junit.Assert.fail(Assert.java:87)
at org.junit.Assert.assertTrue(Assert.java:42)
at org.junit.Assert.assertTrue(Assert.java:53)
at org.mozilla.fenix.ui.robots.BrowserRobot.verifyTabCounter(BrowserRobot.kt:111)
at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$4.invoke(TabbedBrowsingTest.kt:192)
at org.mozilla.fenix.ui.TabbedBrowsingTest$verifyUndoSnackBarTest$4.invoke(TabbedBrowsingTest.kt:191)
at org.mozilla.fenix.ui.robots.BrowserRobotKt.browserScreen(BrowserRobot.kt:658)
at org.mozilla.fenix.ui.TabbedBrowsingTest.verifyUndoSnackBarTest(TabbedBrowsingTest.kt:191)
at java.lang.reflect.Method.invoke(Native Method)
```

**Comment by AndiAJ** [comment]
Fixed and re-enabled in #24727
